crates/wasmparser/src/readers/core/types.rs

```diff
@@ -1491,6 +1536,8 @@ pub enum HeapType {
     ///
     /// Introduced in the function-references proposal.
     Concrete(UnpackedIndex),
+    /// An exact defined type.
```

Comment:
Mind expanding this to indicate that it's part of the custom-descriptors proposal?

Ideally this'd have more comprehensive documentation as to what it actually is (I've no idea what it is myself) but for now if you'd prefer to just delegate to the proposal itself that's fine.